src/components/App/App.js

```diff
@@ -43,17 +43,64 @@ function App() {
       setWinner("It's a tie!")
     }
   }
+  
+  function filterCharactersByNamePublisherAlignment(data, userInput, publisher, alignment){
+    let filteredCharacters = data.characters.filter(character=> {
+
+      if((publisher==='all' && alignment==='all' && character.name.toLowerCase().includes(userInput.toLowerCase()))
+       || (publisher==='all' && alignment==='all' && character.fullName.toLowerCase().includes(userInput.toLowerCase()))
+      ) {return character }
 
-  function filterCharacters(userInput){
-    let filteredCharacters = characters.filter(character=> {
-      return character.name.toLowerCase().includes(userInput.toLowerCase())
+      if(
+        (publisher === character.publisher && alignment==='all' && character.name.toLowerCase().includes(userInput.toLowerCase()))
+       || (publisher === character.publisher && alignment==='all' && character.fullName.toLowerCase().includes(userInput.toLowerCase()))
+       || (publisher === 'all' && alignment===character.alignment && character.name.toLowerCase().includes(userInput.toLowerCase()))
+       || (publisher === 'all' && alignment===character.alignment && character.fullName.toLowerCase().includes(userInput.toLowerCase()))
+       || (publisher === character.publisher && alignment===character.alignment && character.name.toLowerCase().includes(userInput.toLowerCase()))
+       || (publisher === character.publisher && alignment===character.alignment && character.fullName.toLowerCase().includes(userInput.toLowerCase()))
+      ){ return character }
```

Comment:
Nice job refactoring the conditionals for the search and sort functionality! Glad it finally works. we can possible work on shortening this down to make it more readable, but as a long as it works for now, this is good!